tests/receivers/mysql/mysql_test.go

```diff
@@ -28,16 +30,14 @@ import (
 // The reported telemetry may need to be updated (resource metric file changes),
 // They can detect breaking changes or bugs that may have been missed upstream.
 func TestMysqlIntegration(t *testing.T) {
-
-	testutils.AssertAllMetricsReceived(t, "all.yaml", "all_metrics_config.yaml",
-		nil, []testutils.CollectorBuilder{
-			func(collector testutils.Collector) testutils.Collector {
-				return collector.WithEnv(map[string]string{
-					"MYSQLDB_ENDPOINT": "127.0.0.1:3306",
-					"MYSQLDB_USERNAME": "root",
-					"MYSQLDB_PASSWORD": "testpass",
-				})
-			},
-		},
+	testutils.CheckGoldenFile(t, "all_metrics_config.yaml", "all_expected.yaml",
+		pmetrictest.IgnoreScopeVersion(),
+		pmetrictest.IgnoreMetricDataPointsOrder(),
+		pmetrictest.IgnoreResourceMetricsOrder(),
+		pmetrictest.IgnoreScopeMetricsOrder(),
+		pmetrictest.IgnoreMetricsOrder(),
+		pmetrictest.IgnoreMetricValues(),
+		pmetrictest.IgnoreTimestamp(),
+		pmetrictest.IgnoreStartTimestamp(),
```

Comment:
After seeing more tests being moved, I don't think this would be super useful. It seems like most tests have some variation of ignores, so there isn't a single one size fits all set of ignores. We could come up with some kind of groupings, but I don't think this would be worth much time and effort at this point.